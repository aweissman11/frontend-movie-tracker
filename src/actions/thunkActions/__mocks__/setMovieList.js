
export const setMovieList = jest.fn()
// export const getMovieList = jest.fn()