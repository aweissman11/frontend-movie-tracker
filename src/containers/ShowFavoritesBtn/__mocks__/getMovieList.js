export const getMovieList = jest.fn();