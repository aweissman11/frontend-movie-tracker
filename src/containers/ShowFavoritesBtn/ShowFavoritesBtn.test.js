/* eslint-disable */
import React from 'react';
import { shallow } from 'enzyme';
import { ShowFavoritesBtn, mapStateToProps, mapDispatchToProps } from './index'
import { getMovieList } from './__mocks__/getMovieList';

jest.mock('../../actions/thunkActions/movieListThunk');

describe('ShowFavoritesBtn', () => {
  let wrapper;
  let mockShowFavorites;
  let mockSetFavorites;
  const mockFavorites = [
    {
      movie_id: 4
    },
    {
      movie_id: 3
    },
  ];
  
  beforeEach(() => {
    mockShowFavorites = jest.fn();
    mockSetFavorites = jest.fn();

    wrapper = shallow(<ShowFavoritesBtn 
      favorites={mockFavorites}
      setFavorites={mockSetFavorites}
    />)
  });

  it.skip('should match the snapshot MoviesList', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('showfavorites should call setFavorites w/ the right params', () => {
    const expected = [4, 3];
    wrapper.instance().showFavorites();
    expect(mockSetFavorites).toHaveBeenCalledWith(expected);
  });

  // it('should call showFavorites on click', () => {
  //   wrapper.showFavorites = mockShowFavorites;
  //   wrapper.find('button').simulate('click');

  //   console.log(wrapper.instance().showFavorites)
  //   expect(mockShowFavorites).toHaveBeenCalled();
  // });
});

describe('mapStateToProps', () => {
  it('should map the state to props', () => {
    
    const favorites = 
      {"title": "Green Mile"}

    const expected = {favorites}

    const mockState = {
      favorites: favorites    
    }

    const mappedProps = mapStateToProps(mockState)
    expect (mappedProps).toEqual(expected)
  });
});

  describe('mapDispatchToProps', () => {
    it('should map dispatch with getMovieList when setFavorites is called', () => {
      const mockDispatch = jest.fn();
      const movieIds = [100, 45];
      const mappedProps = mapDispatchToProps(mockDispatch)
      mappedProps.setFavorites(movieIds)
      expect(mockDispatch).toHaveBeenCalled()
    });
  });