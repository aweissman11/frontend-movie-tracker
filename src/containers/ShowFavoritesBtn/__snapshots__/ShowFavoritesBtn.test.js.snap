// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowFavoritesBtn should match the snapshot MoviesList 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ShowFavoritesBtn
    favorites={
      Array [
        Object {
          "movie_id": 4,
        },
        Object {
          "movie_id": 3,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "favorites",
      "className": "show-favorites",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": "favorites",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "favorites",
        "className": "show-favorites",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "favorites",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
